### ISSUE TITLE
Install knative failed on Docker for Mac

### ISSUE THREAD

**Issue by juhoffma** [knative/docs#983]
Tried to follow the instructions in: https://github.com/knative/docs/blob/master/install/Knative-with-Docker-for-Mac.md

At the step: 
```
curl -L https://github.com/knative/serving/releases/download/v0.4.0/serving.yaml \
  | sed 's/LoadBalancer/NodePort/' \
  | kubectl apply --filename -
```

## Expected Behavior

The containers should start

## Actual Behavior

```
 buddy@Buddy  ~  curl -L https://github.com/knative/serving/releases/download/v0.4.0/serving.yaml \                                                                 | sed 's/LoadBalancer/NodePort/' \
  | kubectl apply --filename -

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   603    0   603    0     0    677      0 --:--:-- --:--:-- --:--:--   678
100 32897  100 32897    0     0  20597      0  0:00:01  0:00:01 --:--:-- 51401
namespace "knative-serving" created
clusterrole.rbac.authorization.k8s.io "knative-serving-admin" created
clusterrole.rbac.authorization.k8s.io "knative-serving-core" created
serviceaccount "controller" created
clusterrolebinding.rbac.authorization.k8s.io "knative-serving-controller-admin" created
gateway.networking.istio.io "knative-ingress-gateway" created
gateway.networking.istio.io "cluster-local-gateway" created
customresourcedefinition.apiextensions.k8s.io "clusteringresses.networking.internal.knative.dev" created
customresourcedefinition.apiextensions.k8s.io "configurations.serving.knative.dev" created
customresourcedefinition.apiextensions.k8s.io "images.caching.internal.knative.dev" created
customresourcedefinition.apiextensions.k8s.io "podautoscalers.autoscaling.internal.knative.dev" created
customresourcedefinition.apiextensions.k8s.io "revisions.serving.knative.dev" created
customresourcedefinition.apiextensions.k8s.io "routes.serving.knative.dev" created
customresourcedefinition.apiextensions.k8s.io "services.serving.knative.dev" created
service "activator-service" created
service "controller" created
service "webhook" created
deployment.apps "activator" created
service "autoscaler" created
deployment.apps "autoscaler" created
configmap "config-autoscaler" created
configmap "config-controller" created
configmap "config-domain" created
configmap "config-gc" created
configmap "config-istio" created
configmap "config-logging" created
configmap "config-network" created
configmap "config-observability" created
deployment.apps "controller" created
deployment.apps "webhook" created
error: unable to recognize "STDIN": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
```

watching the following command: `kubectl get pods --namespace knative-serving`produced the following output:

```
NAME                          READY     STATUS             RESTARTS   AGE
activator-7c75dffd7b-mvxlm    0/2       PodInitializing    0          31s
autoscaler-7b59bfc4f8-ctqtr   0/2       PodInitializing    0          31s
controller-74588bb9ff-rp859   0/1       CrashLoopBackOff   1          31s
webhook-64587989bf-lwm6p      0/1       CrashLoopBackOff   1          31s
controller-74588bb9ff-rp859   0/1       Error     2         33s
autoscaler-7b59bfc4f8-ctqtr   1/2       Error     0         39s
autoscaler-7b59bfc4f8-ctqtr   1/2       Error     1         40s
autoscaler-7b59bfc4f8-ctqtr   1/2       CrashLoopBackOff   1         41s
controller-74588bb9ff-rp859   0/1       CrashLoopBackOff   2         44s
webhook-64587989bf-lwm6p   0/1       Error     2         45s
activator-7c75dffd7b-mvxlm   1/2       Error     0         46s
activator-7c75dffd7b-mvxlm   1/2       Error     1         47s
activator-7c75dffd7b-mvxlm   1/2       CrashLoopBackOff   1         48s
autoscaler-7b59bfc4f8-ctqtr   1/2       Error     2         56s
webhook-64587989bf-lwm6p   0/1       CrashLoopBackOff   2         58s
controller-74588bb9ff-rp859   0/1       Error     3         59s
controller-74588bb9ff-rp859   0/1       CrashLoopBackOff   3         1m
activator-7c75dffd7b-mvxlm   1/2       Error     2         1m
autoscaler-7b59bfc4f8-ctqtr   1/2       CrashLoopBackOff   2         1m
webhook-64587989bf-lwm6p   0/1       Error     3         1m
activator-7c75dffd7b-mvxlm   1/2       CrashLoopBackOff   2         1m
```

istio namespace looks like this:

```
buddy@Buddy  ~  kubectl get pods --namespace istio-system                                                                                                  NAME                                        READY     STATUS      RESTARTS   AGE
cluster-local-gateway-6c496c6b4-rb4xg       1/1       Running     0          5m
istio-citadel-84fb7985bf-trm64              1/1       Running     0          5m
istio-cleanup-secrets-cx7h5                 0/1       Completed   0          5m
istio-egressgateway-bd9fb967d-gnct9         1/1       Running     0          5m
istio-galley-655c4f9ccd-vmnrk               1/1       Running     0          5m
istio-ingressgateway-688865c5f7-xm6sk       1/1       Running     0          5m
istio-pilot-6cd69dc444-25cps                2/2       Running     0          5m
istio-pilot-6cd69dc444-p7547                2/2       Running     0          5m
istio-pilot-6cd69dc444-tmdjt                2/2       Running     0          5m
istio-policy-6b9f4697d-p9b92                2/2       Running     0          5m
istio-sidecar-injector-8975849b4-zj477      1/1       Running     0          5m
istio-statsd-prom-bridge-7f44bb5ddb-xfhnb   1/1       Running     0          5m
istio-telemetry-6b5579595f-qsnv7            2/2       Running     0          5m
```

## Additional Info
**Additional context**

**Install information:**
 - Platform (GKE, IKS, AKS, etc.): Mac OSX 10.14.3
 - Knative Version: 0.4.0

**Comment by juhoffma** [comment]
```
kubectl -n knative-serving logs controller-74588bb9ff-v6m5c
```
```                                                                              
{"level":"info","ts":"2019-03-12T09:20:10.411Z","caller":"logging/config.go:96","msg":"Successfully created the logger.","knative.dev/jsonconfig":"{\n  \"level\": \"info\",\n  \"development\": false,\n  \"outputPaths\": [\"stdout\"],\n  \"errorOutputPaths\": [\"stderr\"],\n  \"encoding\": \"json\",\n  \"encoderConfig\": {\n    \"timeKey\": \"ts\",\n    \"levelKey\": \"level\",\n    \"nameKey\": \"logger\",\n    \"callerKey\": \"caller\",\n    \"messageKey\": \"msg\",\n    \"stacktraceKey\": \"stacktrace\",\n    \"lineEnding\": \"\",\n    \"levelEncoder\": \"\",\n    \"timeEncoder\": \"iso8601\",\n    \"durationEncoder\": \"\",\n    \"callerEncoder\": \"\"\n  }\n}"}
{"level":"info","ts":"2019-03-12T09:20:10.411Z","caller":"logging/config.go:97","msg":"Logging level set to info"}
{"level":"warn","ts":"2019-03-12T09:20:10.411Z","caller":"logging/config.go:65","msg":"Fetch GitHub commit ID from kodata failed: \"ref: refs/heads/upstream/release-0.4\" is not a valid GitHub commit ID"}
{"level":"fatal","ts":"2019-03-12T09:20:10.421Z","logger":"controller","caller":"controller/main.go:118","msg":"Version check failed: kubernetes version \"v1.10.11\" is not compatible, need at least \"v1.11.0\"","stacktrace":"main.main\n\t/go/src/github.com/knative/serving/cmd/controller/main.go:118\nruntime.main\n\t/root/sdk/go1.12rc1/src/runtime/proc.go:200"}
```

**Comment by juhoffma** [comment]
Seems like I need to update kubernetes in docker for mac

**Comment by yonwon01** [comment]
I got a same problem..

`$ curl -L https://github.com/knative/serving/releases/download/v0.4.0/serving.yaml \   | sed 's/LoadBalancer/NodePort/' \   | kubectl apply --filename -`

`  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   603    0   603    0     0    695      0 --:--:-- --:--:-- --:--:--   695
100 32897  100 32897    0     0  13430      0  0:00:02  0:00:02 --:--:-- 24940
namespace/knative-serving created
clusterrole.rbac.authorization.k8s.io/knative-serving-admin created
clusterrole.rbac.authorization.k8s.io/knative-serving-core created
serviceaccount/controller created
clusterrolebinding.rbac.authorization.k8s.io/knative-serving-controller-admin created
gateway.networking.istio.io/knative-ingress-gateway created
gateway.networking.istio.io/cluster-local-gateway created
customresourcedefinition.apiextensions.k8s.io/clusteringresses.networking.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/configurations.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/podautoscalers.autoscaling.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/revisions.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/routes.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/services.serving.knative.dev created
service/activator-service created
service/controller created
service/webhook created
deployment.apps/activator created
service/autoscaler created
deployment.apps/autoscaler created
configmap/config-autoscaler created
configmap/config-controller created
configmap/config-domain created
configmap/config-gc created
configmap/config-istio created
configmap/config-logging created
configmap/config-network created
configmap/config-observability created
deployment.apps/controller created
deployment.apps/webhook created
error: unable to recognize "STDIN": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"
`

**Comment by schneidersteve** [comment]
I have the same problem with minikube (--vm-driver kvm2 --memory 8192 --cpus 4):

$ curl -L https://github.com/knative/serving/releases/download/v0.4.0/serving.yaml | sed 's/LoadBalancer/NodePort/' | kubectl apply --filename -

`namespace/knative-serving created
clusterrole.rbac.authorization.k8s.io/knative-serving-admin created
clusterrole.rbac.authorization.k8s.io/knative-serving-core created
serviceaccount/controller created
clusterrolebinding.rbac.authorization.k8s.io/knative-serving-controller-admin created
gateway.networking.istio.io/knative-ingress-gateway created
gateway.networking.istio.io/cluster-local-gateway created
customresourcedefinition.apiextensions.k8s.io/clusteringresses.networking.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/configurations.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/images.caching.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/podautoscalers.autoscaling.internal.knative.dev created
customresourcedefinition.apiextensions.k8s.io/revisions.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/routes.serving.knative.dev created
customresourcedefinition.apiextensions.k8s.io/services.serving.knative.dev created
service/activator-service created
service/controller created
service/webhook created
deployment.apps/activator created
service/autoscaler created
deployment.apps/autoscaler created
configmap/config-autoscaler created
configmap/config-controller created
configmap/config-domain created
configmap/config-gc created
configmap/config-istio created
configmap/config-logging created
configmap/config-network created
configmap/config-observability created
deployment.apps/controller created
deployment.apps/webhook created
error: unable to recognize "STDIN": no matches for kind "Image" in version "caching.internal.knative.dev/v1alpha1"`

**Comment by sebgoa** [comment]
"Just" re-issue the command again. It is a race due to the CRDs installation.

**Comment by sebgoa** [comment]
#1036  should fix this for could I believe.

**Comment by schneidersteve** [comment]
I retested successfully with the following minikube configuration and Knative **0.4.1**

1. minikube with --vm-driver kvm2 --memory 16384 --cpus 4
2. kubectl apply --filename https://github.com/knative/serving/releases/download/v0.4.1/istio-crds.yaml && curl -L https://github.com/knative/serving/releases/download/v0.4.1/istio.yaml | sed 's/LoadBalancer/NodePort/' | kubectl apply --filename -
3. kubectl label namespace default istio-injection=enabled
4. curl -L https://github.com/knative/serving/releases/download/v0.4.1/serving.yaml | sed 's/LoadBalancer/NodePort/' | kubectl apply --filename -
5. kubectl apply --filename https://github.com/knative/serving/releases/download/v0.4.1/monitoring.yaml

**Comment by mhamwala** [comment]
@juhoffma @schneidersteve did you manage to get past this error?

**Comment by carieshmarie** [comment]
I think this issue has been resolved with #1036, as mentioned, but please reopen if I'm mistaken.
